robot_calibration/src/capture_features.cpp

```diff
@@ -0,0 +1,304 @@
+// Author: Jack Center
+
+#include <fstream>
+#include <ctime>
+
+#include <ros/ros.h>
+#include <rosbag/bag.h>
+#include <rosbag/view.h>
+#include <rosbag/query.h>
+
+#include <std_msgs/String.h>
+#include <robot_calibration_msgs/CalibrationData.h>
+#include <robot_calibration_msgs/CaptureConfig.h>
+
+#include <robot_calibration/capture/chain_manager.h>
+#include <robot_calibration/capture/feature_finder_loader.h>
+
+#include <camera_calibration_parsers/parse.h>
+#include <robot_calibration/ceres/optimizer.h>
+#include <robot_calibration/camera_info.h>
+#include <robot_calibration/load_bag.h>
+
+#include <boost/foreach.hpp>  // for rosbag iterator
+
+#include <robot_calibration/capture_features.h>
```

Comment:
As it seems you don't plan on making this into a library having a header for it is unnecessary.  You can just put all the functions declared here in a anonymous namespace above the main.  If you want those functions to be in a separate file from the main you would still need this header though.